.gitlab-ci.yml

```diff
@@ -137,10 +137,14 @@ TF_Service::dotNet::build:
   stage: build
   when: always
   script:
-    # Move the built wrapper to the root directory
-    - dotnet publish -c Release "$CI_PROJECT_DIR/TF_Service_dotNet/TouchFree Service/TouchFree Service.sln"
+    - dotnet build -c Release "$CI_PROJECT_DIR/TF_Service_dotNet/TouchFree Service/TouchFree Service.sln"
+
+    # Run unit tests
+    - dotnet test "$CI_PROJECT_DIR/TF_Service_dotNet/TouchFreeTests/TouchFreeTests.csproj"
```

Comment:
We've got a `test` stage in the main CI atm, would it be cleaner to just build the executables here and test them in the test stage? Also, I'm wondering if we could get the unit tests to produce a results xml file, because we could capture that with [GitLab's results](https://docs.gitlab.com/ee/ci/unit_test_reports.html), which may (hopefully) pass the same info to GitHub. It doesn't really matter now, but if we can trivially set up clean test reports now, it may save us a bunch of digging through GitLab consoles to find out what failed later 😅